apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -313,6 +325,28 @@ static void tc_fs_vfs_open(void)
 
 	TC_SUCCESS_RESULT();
 }
+/**
+* @testcase         tc_fs_vfs_fdopen
+* @brief            Open file to do file operation using file descriptor
+* @scenario         Open specific file
+* @apicovered       fs_fdopen
+* @precondition     NA
+* @postcondition    NA
+*/
+static void tc_fs_vfs_fdopen(void)
+{
+	int fd;
+	struct file_struct *fp ;
+	fd = open(VFS_FILE_PATH, O_WROK | O_CREAT);
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	fp = fs_fdopen(fd, O_WROK, NULL);
+	TC_ASSERT_NEQ_CLEANUP("fs_fdopen", fp, NULL, fclose(fp); close(fd));
```

Comment:
if fp is NULL, then fclose(fp) is invalid. I think fclose(fp) can be removed from ASSERT_CLEANUP

Reply:
yes fclose(fp) is not required.